Assets/Scripts/StandardSamples/UI/Achievements/UIAchievementsMenu.cs

```diff
@@ -292,9 +292,12 @@ void DisplayAchievementDefinition(DefinitionV2 definition)
                 "Id: {0}\nUnlocked Display Name: {1}\nUnlocked Description: {2}\nLocked Display Name: {3}\nLocked Description: {4}\nHidden: {5}\n",
                 definition.AchievementId, definition.UnlockedDisplayName, definition.UnlockedDescription, definition.LockedDisplayName, definition.LockedDescription, definition.IsHidden);
 
-            foreach (StatThresholds st in definition.StatThresholds)
+            if (definition.StatThresholds != null)
```

Comment:
Nice catch. Not a suggestion to change - but food for thought is that you can do this in one line by utilizing the null-coalescing operator `??` to provide an empty collection if the value being iterated on is null:

```csharp
foreach (StatThresholds st in definition.StatsThresholds ?? Enumerable.Empty<StatThresholds>)
{
    /* ... */
}
```

But I think you approach in this case is more straightforward and (more importantly) easier to understand.

Reply:
Thank you for sharing! I didn't consider that you could do something like this after the "in" statement as part of a foreach.

A null definition would have other issues here, yea. I _think_ this is the only nullable field inside it.